### PR TITLE
Display an error message if editing by zero on quantity field

### LINKED ISSUE #27866: BO - [Stock] An error message is displayed when editing by zero on quantity field
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add screenshots

An error message is displayed when editing by zero on quantity field (Bad request)
image

PS1784 :x: 

https://user-images.githubusercontent.com/92912932/157465036-d896d915-86ee-412a-83d4-ba2258e2d800.mp4

PS1778 :x:

https://user-images.githubusercontent.com/92912932/157465024-dde9383c-61a0-46c9-99e5-edefce675088.mp4

 

### Expected behavior

The zero is also a number. Make it possible to edit with zero on quantity field.

### Steps to reproduce

1. BO > SELL > Catalog > Stock
2. Click on field to edit quantity
3. Put a `zero` number
See error > An error message is displayed (Bad request)

### PrestaShop version(s) where the bug happened

1.7.7.8 & 1.7.8.4

### PHP version(s) where the bug happened

7.3

### If your bug is related to a module, specify its name and its version

_No response_

### PR DESCRIPTION
**Pull request by Progi1984**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Display an error message if editing by zero on quantity field
| Type?             | bug fix
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #27866
| How to test?      | Cf. #27866

### CODE REVIEW

**Review by Julievrz — DISMISSED** [review]

File: admin-dev/themes/new-theme/js/app/pages/stock/store/actions.ts [hunk 1]
```diff
@@ -169,7 +169,7 @@ export const updateQtyByProductsId = ({commit, state}: {commit: Commit, state: R
     commit(types.UPDATE_PRODUCTS_QTY, res.body);
     EventBus.$emit('displayBulkAlert', 'success');
   }, (error): void => {
-    showGrowl('error', error.statusText);
+    showGrowl('error', error.body && error.body.error ? error.body.error : error.statusText);
```
Comment (NeOMakinG) [inline admin-dev/themes/new-theme/js/app/pages/stock/store/actions.ts]:
```suggestion
    showGrowl('error', error.body?.error ? error.body.error : error.statusText);
```

Comment (NeOMakinG) [reply admin-dev/themes/new-theme/js/app/pages/stock/store/actions.ts]: I guess you can even do something like 
```suggestion
    showGrowl('error', error.body?.error ?? error.statusText);
```

Comment (Progi1984) [reply admin-dev/themes/new-theme/js/app/pages/stock/store/actions.ts]: @NeOMakinG Tested & Fixed.

**Comment by Progi1984** [comment]
@NeOMakinG It's fixed

The wording has already been validated.

**Review by Julievrz — DISMISSED** [review]

**Review by NeOMakinG — DISMISSED** [review]

**Review by ghost — DISMISSED** [review]

**Review by HanaRebaiQA — COMMENTED** [review]
Hello @Progi1984 

Thanks for your PR. The issue is fixed. Here is my checked list : 

-Check with different quantities numbers : OK
-Check with different products types : OK
-Check with multistore : OK
-Check with different languages : OK
-Check with different browsers : OK
-Check the automated tests : OK

image

But, i found two related behaviors different from **PS1787** and already exist on **develop and 8.0.x without this PR** 

**First issue** : When we put zero as a number and click on enter OR click to check it, a green alert is displayed : `Stock successfully updated

develop & 8.0.x : 


https://user-images.githubusercontent.com/92912932/181798155-c07b841a-3d8e-4f6c-9475-52d448ab4c32.mp4

PS1787 :  Without a green alert 

https://user-images.githubusercontent.com/92912932/181798394-025f0171-1e68-4495-b224-fbdf01ff83f2.mp4

**Second issue** : When we select the bulk action to edit quantity and put a zero on the product field, the `Apply new quantity `button is disabled and there is no possibility to check a zero with the new red alert : `Value cannot be zero 

develop & 8.0.x : 

https://user-images.githubusercontent.com/92912932/181799129-00f265d1-37f9-4ac2-b776-49e6e4b57932.mp4



PS1787 : There is the red alert (bad request) which should be fixed in this PR

https://user-images.githubusercontent.com/92912932/181799063-c890bf46-8fbc-49ab-9a8f-ea40f55e69c0.mp4

@MatShir : What do you thins about these two issues ?

Thank You!

`

**Comment by MatShir** [comment]
It's not an ideal result. The button is disabled but you can still press it. @Progi1984 Do you think you can have a look to the issue ?

**Comment by Progi1984** [comment]
@okom3pom & @NeOMakinG Hi, I have fixed bugs for @HanaRebaiQA & @MatShir. Could you review it, please ? The wording has not changed (So please put the label to "Wording :heavy_check_mark:").

**Review by NeOMakinG — DISMISSED** [review]

**Review by matthieu-rolland — DISMISSED** [review]

**Review by HanaRebaiQA — COMMENTED** [review]
Hello @Progi1984 

Thanks for you fix. I have checked again this PR. But, i noticed these two issues : 
- Issue 2 mentioned above is not fixed. The `Apply new quantity` is always disabled when we put zero on bulk edit quantity
image

PS1787 : OK
image


-Following your fix, the alert message `Value cannot be zero ` is not displayed when we put zero on the field like during the previous test.
image

PS1787 : OK
image


Could you please check and confirm for me the expected behavior ?

Thank You!

**Comment by FabienPapet** [comment]
Hello @Progi1984 did you have time to check the issue mentioned above ?

**Comment by Progi1984** [comment]
When i Come back from holidays in one week 😉

**Comment by MatShir** [comment]
I need to test the PR but the behavior seems okay, not be able to click on apply when the value is 0. I'm going to test to see if the user experience is okay

**Comment by Progi1984** [comment]
@MatShir Hi, have you found some time for testing this behavior ? Thanks for advance.

**Comment by MatShir** [comment]
I had issues with my laptop, I will check your PR this week 💯

**Comment by atomiix** [comment]
@MatShir friendly reminder :)

**Comment by Progi1984** [comment]
@MatShir friendly reminder :)

**Comment by MatShir** [comment]
Sounds good to me ! Thanks @Progi1984

**Comment by MatShir** [comment]
The behavior is approved and the other tests as well, should we merge this PR @HanaRebaiQA?

**Comment by HanaRebaiQA** [comment]
Hello @MatShir 

Thanks for your feedback! 
I think this PR should be re-checked again because the test was performed on an old develop version.

Thank You!

**Review by HanaRebaiQA — COMMENTED** [review]
Hello, 

I have checked again this PR. The automatic tests are failed on `BO : Catalog - Stocks ` page in `Bulk edit quantity`
image

It's maybe because the behavior is changed from before as mentioned here in this [comment](https://github.com/PrestaShop/PrestaShop/pull/28917#pullrequestreview-1064994279).

This is the launched test : https://github.com/HanaRebaiQA/testing_pr/actions/runs/3233281251 

ping @PrestaShop/qa-automation Could you please check it ? 

Thank You!

**Comment by Progi1984** [comment]
@matthieu-rolland, @NeOMakinG I fixed errors detected by UI Tests (and @HanaRebaiQA ) in the last commit (04e2c33). If I can have a new review, it would be coooooooooooool.

**Review by NeOMakinG — APPROVED** [review]

**Review by atomiix — APPROVED** [review]

**Review by HanaRebaiQA — APPROVED** [review]
Hello @Progi1984 

Thanks for the fixed issue. I have checked again this PR. 
It's OK :heavy_check_mark: 

Here is my checked list : 
Check the automated tests: OK
Check the `zero` with different options (bulk action, simple field) : OK
Check the different other numbers: OK
Check with multistore: OK
Check with different languages (RTL and no RTL) : OK
Check with different browsers (Chrome and Firefox) : OK


image
image

Thank You!

**Pull request merged**